apps/rush-lib/src/logic/policy/LockfilePolicy.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as colors from 'colors';
+import * as os from 'os';
+import { InternalError } from '@microsoft/node-core-library';
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { AlreadyReportedError } from '../../utilities/AlreadyReportedError';
+import { IPolicyValidatorOptions } from './PolicyValidator';
+import { BaseShrinkwrapFile } from '../base/BaseShrinkwrapFile';
+import { ShrinkwrapFileFactory } from '../ShrinkwrapFileFactory';
+
+export class LockfilePolicy {
+  public static validate(rushConfiguration: RushConfiguration, options: IPolicyValidatorOptions): void {
+    // If allowShrinkwrapUpdates is true, then we're updating the shrinkwrap
+    // and cannot validate the hash
+    if (options.allowShrinkwrapUpdates) {
+      return;
+    }
+
+    if (!LockfilePolicy.isEnabled(rushConfiguration)) {
+      // If the option wasn't provided
+      console.log(
+        colors.cyan(
+          'Ignoring lockfile validation because it was not enabled or is not supported ' +
+          'with the specified package manager.'
```

Comment:
Is this printed every time `rush install` is run?  It seems like useless noise.  Also, the feature is scoped under `PnpmOptionsConfiguration` so there is no expectation for it to work with other package managers.